landlab/components/dimensionless_discharge/dimensionless_discharge.py

```diff
@@ -0,0 +1,196 @@
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "dimensionless_discharge_above_threshold": {
+            "dtype": bool,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "flux": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m/s",
+            "mapping": "node",
+            "doc": "flux in each stream segment",
+        },
+        "d50": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "soil grain size average in stream segment",
+        },
+        "dem_values": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "elevation stream segment",
+        },
+    }
+
+    def __init__(self, grid, soil_density=1330, water_density=997.9, C=12.0, N=0.85):
+        """Initialize the DimensionlessDischarge.
+
+        Parameters
+        ----------
+        soil_density : float, optional (defaults to 1330)
+            density of soil (kg/m^3)
+        water_density : float, optional (defaults to 997.9)
+            density of water (kg/m^3)
+        C : float, optional (defaults to 12.0)
+            Numerator of the debris flow threshold equation; Empirically
+            derived constant (See Tang et al. 2019)
+        N : float, optional (defaults to 0.85)
+            Exponent for slope in the denominator of the debris flow
+            threshold equation; Empirically derived constant (See Tang
+            et al. 2019)
+        """
+
+        super().__init__(grid)
+
+        # Store parameters
+        self._soil_density = soil_density
+        self._C = [C] * grid.number_of_nodes
+        self._N = [N] * grid.number_of_nodes
+        # change DEM values into slope of a stream segment
+        self._stream_slopes = grid.calc_slope_at_node(elevs="dem_values")
+        self.water_density = water_density
+        self.gravity = 9.8
+
+        # set threshold values for each segment
+        _ = self.grid.add_zeros("node", "dimensionless_discharge")
+        _ = self.grid.add_zeros("node", "dimensionless_discharge_above_threshold")
+        self.grid.at_node["dimensionless_discharge_above_threshold"] = np.array(
+            [[False] * self.grid.number_of_nodes]
+        )
+        _ = self.grid.add_zeros("node", "dimensionless_discharge_threshold_value")
+        self.grid.at_node["dimensionless_discharge_threshold_value"] = self._C / (
+            self._stream_slopes ** self._N
+        )
```

Comment:
@mcflugen I think that C and N do need to be stored, since the threshold value should also be calculated in run one step (because it depends on the slopes, which should be calculated in run one step).

Reply:
@kbarnhart Yes, definitely. I wrote this comment before I realized that the slopes would have to be calculated within *run_one_step*.

Reply:
The *dimensionless_discharge_threshold_value* is neither an input or an output field. Perhaps it should just be an attribute of the class?
```python
    self._dimensionless_discharge_threshold_value = C / self._stream_slopes ** N
```
Or, if this is being moved into *run_one_step*, it can just be a local variable.
```python
    dimensionless_discharge_threshold_value = self._c / stream_slopes ** self._n
```

Reply:
I think its meant to be an output field...

Reply:
Makes sense.

@slundell123 if that's the case, please add this field to the class's *_info* dictionary. I suggest also renaming it to just *dimensionless_discharge_threshold*.

Reply:
I have added the field to the _info dictionary. It is used to determine if the dimensionless discharge value would exceed the threshold and  produce a debris flow, but it is possible that someone would like access to this value as an output.